Resources/public/js/ezrecommendation.js

```diff
@@ -0,0 +1,77 @@
+// register HandleBars helpers when library is loaded
+
+document.addEventListener('DOMContentLoaded', function() {
+    Handlebars.registerHelper('each', function(context, options) {
+        var result = '';
+        for(var i = 0, j = context.length; i < j; i++) {
+            result = result + options.fn(context[i]);
+        }
+        return result;
+    });
+
+    Handlebars.registerHelper('formatDate', function(timestamp) {
+        var fixedTimestamp = (timestamp + '000') * 1;
+        var t = new Date(fixedTimestamp);
+        return t.toLocaleDateString();
+    });
+
+    Handlebars.registerHelper('formatTime', function(timestamp) {
+        var fixedTimestamp = (timestamp + '000') * 1;
+        var t = new Date(fixedTimestamp);
+        return t.toLocaleTimeString();
+    });
+});
+
+// EzRecoRestClient class
+
+var EzRecoRestClient = function (config) {
+    this.config = config;
+};
+
+EzRecoRestClient.prototype.fetchRecommendations = function (responseCallback, targetId, templateId) {
+    var xmlhttp = new XMLHttpRequest();
+    var jsonResponse;
+
+    xmlhttp.onreadystatechange = function () {
+        if (xmlhttp.readyState === XMLHttpRequest.DONE) {
+            if (xmlhttp.status == 200) {
+                jsonResponse = JSON.parse(xmlhttp.response);
+            } else {
+                jsonResponse = new Array();
+                jsonResponse['status'] = 'fail';
+            }
+            responseCallback(jsonResponse, targetId, templateId);
+        }
+    };
+
+    xmlhttp.open('GET', window.location.protocol + '//' + window.location.host + this.config['restUrl'] + '?locationId=' + this.config['locationId'] + '&scenarioId=' + this.config['scenarioId'] + '&limit=' + this.config['limit'], true);
```

Comment:
no need to concatenate the protocol, the host, ...
`xmlhttp.open('GET', this.config['restUrl'] + othersParameter);` should do the trick.